ui/app/components/ui/dialog/index.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import c from 'classnames'
+
+export default function Dialog (props) {
+  const { children, type, className, onClick } = props
+  return (
+    <div
+      className={c('dialog', className, {
+        'dialog--message': type === 'message',
+        'dialog--error': type === 'error',
+        'dialog--warning': type === 'warning',
+      })}
+      onClick={onClick}
+    >
+      { children }
+    </div>
+  )
+}
+
+Dialog.propTypes = {
```

Comment:
Can we make this a class and have these propTypes as a static property?

Reply:
Actually, I don't know about this. Have we decided to use class components instead of functional components in all cases? There are a handful of other functional components in the `dialog/index.js` directory, and I sort of like the idea of using them where/when its practical.

Reply:
I agree. I prefer functional components. They're also supposedly faster as of React v16.

Also the React team has indicated that they're planning to deprecate classes eventually, though they'll continue to be supported possibly indefinitely. But functional components plus hooks are the path they're recommending.

Reply:
Sure, this isn't a hill I care to die on